internal/parse/parser.go

```diff
@@ -111,16 +119,36 @@ func (p *Parser) add(part queryPart) {
 
 // Parse takes an input string and parses the input and output parts. It returns
 // a pointer to a ParsedExpr.
-func (p *Parser) Parse(input string) (*ParsedExpr, error) {
+func (p *Parser) Parse(input string) (expr *ParsedExpr, err error) {
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf("cannot parse expression: %s", err)
+		}
+	}()
 	p.init(input)
+
 	for {
 		p.partStart = p.pos
-		if p.pos == len(p.input) {
+
+		if ip, ok, err := p.parseInputExpression(); err != nil {
+			return nil, err
+		} else if ok {
+			p.add(ip)
+
+		} else if sp, ok, err := p.parseStringLiteral(); err != nil {
+			return nil, err
+		} else if ok {
+			p.add(sp)
+
+		} else if p.pos == len(p.input) {
 			break
+		} else {
+			// If nothing above can be parsed we advance the parser.
+			p.pos++
 		}
-		p.pos++
 	}
-	// Add any remaining unparsed string input to the parser.
+	// Add any remaining uparsed string input to the parser
```

Comment:
Looks like the comment was corrupted here. The former line was right.

Reply:
fixed